connectors-e2e-test/connectors-e2e-test-agentic-ai/src/test/java/io/camunda/connector/e2e/agenticai/aiagent/BaseAiAgentTest.java

```diff
@@ -73,27 +74,18 @@ protected ZeebeTest createProcessInstance(
     return deployModel(updatedModel).createInstance(variables);
   }
 
-  protected AgentResponse getAgentResponse(ZeebeTest zeebeTest) throws JsonProcessingException {
-    final var agentVariableSearchResult =
-        new CamundaDataSource(camundaClient)
-                .findGlobalVariablesByProcessInstanceKey(
-                    zeebeTest.getProcessInstanceEvent().getProcessInstanceKey())
-                .stream()
-                .filter(variable -> variable.getName().equals("agent"))
-                .findFirst()
-                .orElseThrow(() -> new RuntimeException("Agent variable 'agent' not found"));
-
-    if (agentVariableSearchResult.isTruncated()) {
-      final var agentVariable =
-          camundaClient
-              .newVariableGetRequest(agentVariableSearchResult.getVariableKey())
-              .send()
-              .join();
-
-      return objectMapper.readValue(agentVariable.getValue(), AgentResponse.class);
-    }
-
-    return objectMapper.readValue(agentVariableSearchResult.getValue(), AgentResponse.class);
+  protected void assertAgentResponse(
+      ZeebeTest zeebeTest, ThrowingConsumer<AgentResponse> assertions) {
+    CamundaAssert.assertThat(zeebeTest.getProcessInstanceEvent())
+        .hasVariableSatisfies(
+            AGENT_RESPONSE_VARIABLE,
+            Map.class,
```

Comment:
Using `AgentResponse` here is not possible - it breaks as soon as the response contains documents as the used ObjectMapper does not include the document module. So we use a map here instead and transform to `AgentResponse` with an ObjectMapper configured to support document references.